Undefined reference to void Utilities::MPI::internal::all_reduce<bool>()

I compiled ASPECT in the release mode but got the following error:
`ld.bfd: /work2/08254/jiaqi2/frontera/dealii/installed_intel_O3/lib/libdeal_II.so.9.4.0-pre: undefined reference to void dealii::Utilities::MPI::internal::all_reduce<bool>(int const&, dealii::ArrayView<bool const, dealii::MemorySpace::Host> const&, int const&, dealii::ArrayView<bool, dealii::MemorySpace::Host> const&)`

The function is here: https://github.com/dealii/dealii/blob/66dd63460d0717c589a54302bd0ff80f5a94c8c7/source/base/mpi.inst.in#L100-L111

Information about the configuration:
```
#  ASPECT configuration:
#        ASPECT_VERSION:            2.4.0-pre
#        GIT REVISION:              edcb0fa34 (main)
#        CMAKE_BUILD_TYPE:          Release
#
#        DEAL_II VERSION:           9.4.0-pre
#        CMAKE_CXX_COMPILER:        Intel 19.1.0.20200306 on platform Linux x86_64
#                                   /opt/intel/compilers_and_libraries_2020.4.304/linux/mpi/intel64/bin/mpiicpc
#        _WITH_CXX14:               ON
#        _WITH_CXX17:               FALSE
#        _MPI_VERSION:              3.1
#        _WITH_64BIT_INDICES:       ON
```




<!--
Thanks for reporting an issue with deal.II! This HTML comment won't be visible
in the issue tracker but is present here to help you write a better description
of the problem you encountered.

If you have questions about deal.II, please post them on the mailing list:

   https://groups.google.com/g/dealii

Otherwise:
- Do you encounter a compilation failure? Please attach the detailed.log file
  present in the build directory here. Any additional compiler messages will also
  be helpful!

- Does something crash (e.g., an example) that shouldn't? Please include any relevant
  terminal output as well as the detailed.log file generated in your build
  directory (should you have one).

- Is there a missing feature you want to implement? Please briefly describe it and
  the problem it solves.

- Did you find some other issue with deal.II? Summarize the problem here.
-->

--- a/source/base/mpi.cc
+++ b/source/base/mpi.cc
@@ -1197,6 +1197,42 @@ namespace Utilities
 
 #ifndef DOXYGEN
     // explicit instantiations
+
+    // booleans aren't in MPI_SCALARS
+    template bool
+    reduce(const bool &,
+           const MPI_Comm &,
+           const std::function<bool(const bool &, const bool &)> &,
+           const unsigned int);
+
+    template std::vector<bool>
+    reduce(const std::vector<bool> &,
+           const MPI_Comm &,
+           const std::function<std::vector<bool>(const std::vector<bool> &,
+                                                 const std::vector<bool> &)> &,
+           const unsigned int);
+
+    template bool
+    all_reduce(const bool &,
+               const MPI_Comm &,
+               const std::function<bool(const bool &, const bool &)> &);
+
+    template std::vector<bool>
+    all_reduce(
+      const std::vector<bool> &,
+      const MPI_Comm &,
+      const std::function<std::vector<bool>(const std::vector<bool> &,
+                                            const std::vector<bool> &)> &);
+
+    // We need an explicit instantiation of this for the same reason as the
+    // other types described in mpi.inst.in
+    template void
+    internal::all_reduce<bool>(const MPI_Op &,
+                               const ArrayView<const bool> &,
+                               const MPI_Comm &,
+                               const ArrayView<bool> &);
+
+
     template bool
     logical_or<bool>(const bool &, const MPI_Comm &);
 

